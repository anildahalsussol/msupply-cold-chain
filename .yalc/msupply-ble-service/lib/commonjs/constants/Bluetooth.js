'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true,
});
exports.BLUE_MAESTRO = exports.BLUETOOTH = void 0;
let BLUETOOTH;
exports.BLUETOOTH = BLUETOOTH;

(function (BLUETOOTH) {
  BLUETOOTH['UART_SERVICE_UUID'] = '6e400001-b5a3-f393-e0a9-e50e24dcca9e';
  BLUETOOTH['READ_CHARACTERISTIC_UUID'] = '6e400002-b5a3-f393-e0a9-e50e24dcca9e';
  BLUETOOTH['WRITE_CHARACTERISTIC_UUID'] = '6e400003-b5a3-f393-e0a9-e50e24dcca9e';
  BLUETOOTH[(BLUETOOTH['SCAN_MODE_LOW_LATENCY'] = 2)] = 'SCAN_MODE_LOW_LATENCY';
})(BLUETOOTH || (exports.BLUETOOTH = BLUETOOTH = {}));

const BLUE_MAESTRO = {
  COMMANDS: {
    BLINK: '*blink',
    DOWNLOAD: '*logall',
    INFO: '*info',
    UPDATE_LOG_INTERVAL: '*lint',
    DISABLE_BUTTON: '*bd',
  },
  MANUFACTURER_ID: 307,
  DELIMITER_A: 11776,
  DELIMITER_B: 11308,
  TEMPERATURE_DIVISOR: 10.0,
};
exports.BLUE_MAESTRO = BLUE_MAESTRO;
//# sourceMappingURL=Bluetooth.js.map
