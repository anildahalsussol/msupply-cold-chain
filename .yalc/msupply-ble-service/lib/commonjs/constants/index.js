"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BLUETOOTH", {
  enumerable: true,
  get: function () {
    return _Bluetooth.BLUETOOTH;
  }
});
Object.defineProperty(exports, "BLUE_MAESTRO", {
  enumerable: true,
  get: function () {
    return _Bluetooth.BLUE_MAESTRO;
  }
});

var _Bluetooth = require("./Bluetooth");
//# sourceMappingURL=index.js.map