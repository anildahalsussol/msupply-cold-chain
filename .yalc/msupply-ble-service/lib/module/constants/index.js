export { BLUETOOTH } from './Bluetooth';
export { BLUE_MAESTRO } from './Bluetooth';
//# sourceMappingURL=index.js.map
