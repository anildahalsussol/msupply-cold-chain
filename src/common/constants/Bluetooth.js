export const BLUETOOTH = {
  UART_SERVICE_UUID: '6e400001-b5a3-f393-e0a9-e50e24dcca9e',
  READ_CHARACTERISTIC_UUID: '6e400002-b5a3-f393-e0a9-e50e24dcca9e',
  WRITE_CHARACTERISTIC_UUID: '6e400003-b5a3-f393-e0a9-e50e24dcca9e',
};
