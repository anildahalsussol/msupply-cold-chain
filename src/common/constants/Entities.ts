export enum ENTITIES {
  SENSOR = 'Sensor',
  TEMPERATURE_LOG = 'TemperatureLog',
  TEMPERATURE_BREACH = 'TemperatureBreach',
  TEMPERATURE_BREACH_CONFIGURATION = 'TemperatureBreachConfiguration',
  SENSOR_LOG = 'SensorLog',
  SETTING = 'Setting',
  SYNC_LOG = 'SyncLog',
  SYNC_CONFIG = 'SyncConfig',
}
