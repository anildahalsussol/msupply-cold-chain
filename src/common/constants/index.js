export { TEMPERATURE_SYNC_STATE, PASSIVE_TEMPERATURE_SYNC_STATE } from './TemperatureSyncState';
export { REDUCER } from './Reducer';
export { MILLISECONDS } from './Milliseconds';
export { DEPENDENCY, BLUETOOTH_SERVICE } from './Dependency';
export { ENTITIES } from './Entities';
export {
  BLUETOOTH_STATE,
  DANGEROUS_BATTERY_LEVEL,
  PERMISSION,
  PERMISSION_STATE,
} from './NativeDevice';
export {
  DATABASE_UTIL_ERROR,
  BLUETOOTH_SERVICE_ERROR,
  SERVICE_LOCATOR_ERROR,
  UI_ASSERTIONS,
} from './Error';

export { COLOUR } from './Colour';
export { FONT } from './Font';
export { STYLE } from './Style';
export { ICON } from './Icon';
export { FORMAT } from './Format';
export { CHART } from './Chart';
export { NAVIGATION } from './Navigation';
export { BLUETOOTH } from './Bluetooth';
export { SETTING } from './Setting';
