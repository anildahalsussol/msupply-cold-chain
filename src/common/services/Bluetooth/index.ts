export { BleService } from './BleService';
export { DevBleManager } from './DevBleManager';
