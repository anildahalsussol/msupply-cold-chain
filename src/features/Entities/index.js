export {
  BreachConfigurationReducer,
  BreachConfigurationAction,
  BreachConfigurationSelector,
  BreachConfigurationSaga,
  BreachConfigurationManager,
} from './BreachConfiguration';

export { SensorReducer, SensorAction, SensorSelector, SensorSaga, SensorManager } from './Sensor';

export {
  SettingReducer,
  SettingAction,
  SettingSelector,
  SettingSaga,
  SettingManager,
} from './Setting';

export { EntitiesReducer, EntitiesSaga } from './EntitiesSlice';
