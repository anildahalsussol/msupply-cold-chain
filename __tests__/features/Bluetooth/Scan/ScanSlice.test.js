import { expectSaga } from 'redux-saga-test-plan';
import { getContext } from 'redux-saga/effects';
import { ScanReducer, ScanInitialState, ScanAction, ScanSaga } from '~features/Bluetooth';
import { DEPENDENCY } from '../../../../src/common/constants';
import { SensorAction } from '../../../../src/features/Entities';

describe('ScanAction', () => {
  it('has matching snapshots', () => {
    const actions = Object.values(ScanAction).map(action => action());
    expect(actions).toMatchSnapshot();
  });
});

describe('ScanReducer', () => {
  it('stopFail', () => {
    const newState = ScanReducer(
      { ...ScanInitialState, foundSensors: ['A'] },
      SensorAction.createSuccess('id', { macAddress: 'A' })
    );
    expect(newState).toEqual({ foundSensors: [], isScanning: false });
  });
  it('stopFail', () => {
    const newState = ScanReducer(ScanInitialState, ScanAction.startFail());
    expect(newState).toEqual({ foundSensors: [], isScanning: false });
  });
  it('stopSuccess', () => {
    const newState = ScanReducer(ScanInitialState, ScanAction.startFail());
    expect(newState).toEqual({ foundSensors: [], isScanning: false });
  });
  it('startFail', () => {
    const newState = ScanReducer(ScanInitialState, ScanAction.startFail());
    expect(newState).toEqual({ foundSensors: [], isScanning: false });
  });
  it('startSuccess', () => {
    const newState = ScanReducer(ScanInitialState, ScanAction.startSuccess());
    expect(newState).toEqual({ foundSensors: [], isScanning: true });
  });
});

describe('ScanSaga', () => {
  it('', () => {
    const btService = { scanForSensors: callback => callback({ id: 'A' }), stopScan: () => {} };
    const depsLocator = { get: () => btService };
    return expectSaga(ScanSaga.root)
      .provide([[getContext(DEPENDENCY.LOCATOR), depsLocator]])
      .dispatch(ScanAction.tryStart())
      .put(ScanAction.foundSensor('A'))
      .run();
  });
});
