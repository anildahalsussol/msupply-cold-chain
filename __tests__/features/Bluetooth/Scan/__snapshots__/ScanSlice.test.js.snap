// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanAction has matching snapshots 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "scan/tryStart",
  },
  Object {
    "payload": undefined,
    "type": "scan/startSuccess",
  },
  Object {
    "payload": undefined,
    "type": "scan/startFail",
  },
  Object {
    "payload": undefined,
    "type": "scan/tryStop",
  },
  Object {
    "payload": undefined,
    "type": "scan/stopSuccess",
  },
  Object {
    "payload": undefined,
    "type": "scan/stopFail",
  },
  Object {
    "payload": Object {
      "mac": undefined,
    },
    "type": "scan/foundSensor",
  },
]
`;
